app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,182 @@
+        val number = findViewById<EditText>(R.id.numberEt)
+        val pincode = findViewById<EditText>(R.id.pincodeEt)
+        val address = findViewById<EditText>(R.id.addressEt)
+        val validate = findViewById<Button>(R.id.buttonLabel)
+        val cancel=findViewById<Button>(R.id.cancelButton)
+        val confirm=findViewById<Button>(R.id.confirmButton)
+
+        validate.setOnClickListener() {
+            val isAllFieldsChecked = checkAllFields(userName,email,number,pincode,address)
+            if (isAllFieldsChecked) {
+                val validateResult=validateAllFields(userName,email,number,pincode,address)
+                if (validateResult) {
+                    visibility=false
+                    modifyVisibility(userName,email,number,pincode,address,validate,cancel,confirm,visibility)
+
+                }
+            }
+            confirm.setOnClickListener() {
+                if (isAllFieldsChecked) {
+                    val validateResult=validateAllFields(userName,email,number,pincode,address)
+                    if (validateResult) {
+                        val intent = Intent(this, DisplayUserInfoActivity::class.java)
+                        intent.putExtra("USERNAME", userName.text.toString())
+                        intent.putExtra("EMAIL", email.text.toString())
+                        intent.putExtra("PHONENUMBER", number.text.toString())
+                        intent.putExtra("PINCODE", pincode.text.toString())
+                        intent.putExtra("ADDRESS", address.text.toString())
+                        startActivity(intent)
+                    }
+                }
+            }
+            cancel.setOnClickListener() {
+                visibility=true
+                modifyVisibility(userName, email, number, pincode, address,validate,cancel, confirm, visibility)
+            }
+        }
+
+    }
+
+    fun validateAllFields(
+        userName: EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText
+    ): Boolean {
+
+        if (!isValidEmail(email.text.toString())) {
+            callToast( "Invalid Email Address")
+            return false
+        }
+        if (!isValidPinCode(pincode.text.toString())) {
+            callToast("Invalid Pincode")
+            return false
+        }
+        if (!isValidPhone(number.text.toString())) {
+            callToast("Invalid Phone Number")
+            return false
+        }
+        if(!isValidUserName(userName.text.toString())){
+            callToast("Invalid User Name")
+            return false
+        }
+        return true
+    }
+
+    private fun callToast(str:String){
+        Toast.makeText(applicationContext, str,
+            Toast.LENGTH_SHORT).show()
+    }
+
+    private fun checkAllFields(
+        userName: EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText
```

Comment:
Normally it is better pass values instead of Views. As a improvement we can pass a data class which contains user details ( all individual fields will be members of that data class)